docroot/modules/custom/sitenow_advanced_webform/src/Plugin/WebformHandler/ProspectorRemotePostWebformHandler.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\sitenow_advanced_webform\Plugin\WebformHandler;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\webform\Plugin\WebformHandlerBase;
+use Drupal\webform\WebformSubmissionInterface;
+use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Webform submission remote post handler.
+ *
+ * @WebformHandler(
+ *   id = "prospector_remote_post",
+ *   label = @Translation("Prospector Remote Post"),
+ *   category = @Translation("External"),
+ *   description = @Translation("Posts webform submissions to a URL."),
+ *   cardinality = \Drupal\webform\Plugin\WebformHandlerInterface::CARDINALITY_UNLIMITED,
+ *   results = \Drupal\webform\Plugin\WebformHandlerInterface::RESULTS_PROCESSED,
+ *   submission = \Drupal\webform\Plugin\WebformHandlerInterface::SUBMISSION_OPTIONAL,
+ *   tokens = TRUE,
+ * )
+ */
+class ProspectorRemotePostWebformHandler extends WebformHandlerBase {
+
+  /**
+   * The HTTP client to fetch the feed data with.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->httpClient = $container->get('http_client');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::submitConfigurationForm($form, $form_state);
+    $this->applyFormStateToConfiguration($form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function prepareForm(
+    WebformSubmissionInterface $webform_submission,
+    $operation,
+    FormStateInterface $form_state,
+  ) {
+    // @todo This doesn't seem quite right because it might show to an end user.
```

Comment:
Should this be warning messages in `buildConfigurationForm` and be validated in `submitConfigurationForm`?

Reply:
Yeah, I think that makes sense for the message. I'm not sure about the validation. Like would we make it so they couldn't save the handler if there wasn't a UUID? It might just need to be informational.

Reply:
I don't think it is too bold to require values for those before the handler is set and saved. We do it for the thank you block:
<img width="483" alt="Screenshot 2024-10-31 at 10 29 47 AM" src="https://github.com/user-attachments/assets/a9d5cc14-4eec-46f9-b491-2a0d734fda55">

Reply:
To be honest, I really don't like the UX of the above. Someone can add the block only to find out that they can't use it. Preventing it from showing in the list of blocks and providing a help message somewhere else seem like a better pattern.

I think allowing the handler to be added and then adding a message either on the configuration form and/or the handlers page that it won't submit to the app until that extra step is taken is an okay user experience. Its annoying to be at the point where you are trying to submit a form, you have to leave to go do something (add the UUID to a different config form) and then come back to add the handler again.

Reply:
i get what you are saying. i think it is a balance of working within/leverage what already exists/is available.

Reply:
`buildConfigurationForm` doesn't seem to work or least I'm not sure how to add a message that gets displayed on the form since it is loaded into the sidebar via AJAX. The message about the Site UUID gets displayed after the page refreshes. Probably there is a way to set this correctly via the `$form` or `$formState`. Either way, it should possibly also be displayed on the handler page if the handler is enabled.

Reply:
ah. i think we encountered that with pardot as well and just did a form api markup element: https://github.com/uiowa/uiowa/blob/316d7964ab5bf8724e02cd3f5702fba67b048dce/docroot/sites/tippie.uiowa.edu/modules/tippie_core/src/Plugin/WebformHandler/PardotRemotePostWebformHandler.php#L42

Reply:
```
if (!$site_uuid) {
        $form['missing_uuid'] = [
            '#markup' => $this->t('<strong>Warning:</strong> The site UUID is missing. Please configure it...
...
```

Reply:
I added something but I'm not figuring out how to position it at the top of the form. I wondering if this is enough now and the message can be improved in future work?